docs/ingress-tls.md

```diff
@@ -0,0 +1,285 @@
+# Using Secrets Store CSI and Azure Key Vault Provider to Enable NGINX Ingress Controller with TLS
+
+This guide demonstrates steps required to setup Secrets Store CSI driver and Azure Key Vault Provider to enable applications to work with NGINX Ingress Controller with TLS certificates stored in Key Vault.
+For more information on securing an Ingress with TLS, refer to [this guide](https://kubernetes.io/docs/concepts/services-networking/ingress/#tls)
+
+Importing the ingress TLS certificate to the cluster can be included in the following deployments:
+
+* Ingress Controller - Ingress deployment is modified to declare and mount the csi provider volume. The secret is imported when Ingress pods are created and the application's pods have no access to the TLS certificate. This scenario fits teams where Ingress and its related certificates are managed centrally by an IT unit.
```

Comment:
The CSI driver creates the Kubernetes secrets in the same namespace as the application pod requesting volume mount. So in this case the ingress resource will also need to be created in the same namespace as ingress controller as Kubernetes doesn't allow referencing secret from different namespace. Is this something users currently do?

Reply:
If the ingress and related certificates are managed by the IT Unit/cluster admin does this mean they're also responsible for creating and managing the ingress resources?

Reply:
1. yes, this is something some customers do, and is documented. for instance here in the nginx ingress docs: https://docs.nginx.com/nginx-ingress-controller/installation/running-multiple-ingress-controllers/#running-multiple-nginx-ingress-controllers and here in the ms-pnp AKS security best practices guidelines: https://github.com/mspnp/aks-secure-baseline/blob/main/workload/traefik.yaml
2. added a note that in this scenario the ingress is namespace/workload specific.